include/cuco/detail/open_addressing_impl.cuh

```diff
@@ -135,6 +137,55 @@ class open_addressing_impl {
     this->clear_async(stream);
   }
 
+  /**
+   * @brief Constructs a statically-sized open addressing data structure with the number of elements
+   * to insert, sentinel values, the desired load factor and CUDA stream
+   *
+   * @note The actual capacity depends on the given `n`, the probing scheme, CG size, the desired
+   * load factor and the window size and it is computed via the `make_window_extent` factory. Insert
+   * operations will not automatically grow the container. Attempting to insert more unique keys
+   * than the capacity of the container results in undefined behavior.
+   * @note Any `*_sentinel`s are reserved and behavior is undefined when attempting to insert
+   * this sentinel value.
+   * @note If a non-default CUDA stream is provided, the caller is responsible for synchronizing the
+   * stream before the object is first used.
+   * @note Static extent will become dynamic extent by invoking this constructor.
+   *
+   * @throw If the desired occupancy is no bigger than zero
+   * @throw If the desired occupancy is larger than one
+   *
+   * @param n The number of elements to insert
+   * @param desired_load_factor The desired load factor of the container, e.g., 0.5 implies a 50%
+   * load factor
+   * @param empty_key_sentinel The reserved key value for empty slots
+   * @param empty_slot_sentinel The reserved slot value for empty slots
+   * @param pred Key equality binary predicate
+   * @param probing_scheme Probing scheme
+   * @param alloc Allocator used for allocating device storage
+   * @param stream CUDA stream used to initialize the data structure
+   */
+  constexpr open_addressing_impl(Extent n,
+                                 double desired_load_factor,
```

Comment:
Can we have strong type for `desired_load_factor`? Such as
```
struct load_factor_type : double {};
```

Reply:
Using strong type in general is a good practice to avoid misuse (or passing the wrong parameter). Here a strong type would be (IMO) another extra indirection layer since users need to realize that `load_factor_type` is a `double` or can be converted to `double` and then assign a reasonable value to it. With our runtime check for the load factor to be a real number larger than 0 and smaller than 1, the chance of misuse is already low so the benefits of using a strong type seem not to outweigh the costs.

Reply:
I agree with @PointKernel. The load factor is already a positional argument and cannot be confused with any of the subsequent arguments, as they are strong types. I'm ok with using a plain `double` here.

Reply:
I'd be a fan of a strong type here, but floating point strong types are a hassle to implement because you can't do the enum class trick nor can you inherit from `double` as Nghia showed. So it's likely not worth the effort here.